http2_server.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+import socket
+import email.utils
+import mimetypes
+import os
+
+
+ROOT = os.path.join(os.getcwd(), 'webroot')
+
+
+def response_ok(resolved_uri):
+    """Returns response code with date and headers"""
+    response_code = "HTTP/1.1 200 OK"
+    headers = "Content-Type: {}".format(resolved_uri[0])
+    date = "Date: {}".format(email.utils.formatdate(usegmt=True))
+    response = "{}\r\n{}\r\n{}\r\n\r\n{}".format(
+        response_code, date, headers, resolved_uri[1])
+    return response
+
+
+def response_error(code, reason):
+    """formats errors raised to be returned to client"""
+    response_code = "HTTP/1.1 {} {}".format(code, reason)
+    date = "Date: {}".format(email.utils.formatdate(usegmt=True))
+    headers = "Content-Type: text/plain"
+    response = u"{}\r\n{}\r\n{}\r\n\r\n{}".format(response_code, date, headers, reason)
+    return response
+
+
+def parse_request(request):
+    """parse request and return URI, errors raised for invalid protocols
+    and requests
+    """
+    request = request.split('\r\n')
+    first_line = request[0].split(' ')
+    if first_line[0] != "GET":
+        raise RequestError(405, "Method Not Allowed")
+    elif first_line[2] != "HTTP/1.1":
+        raise RequestError(505, "HTTP Version Not Supported")
+    else:
+        return first_line[1]
+
+
+def resolve_uri(uri):
+    """returns tuple of (content, body), if directory, returns listing of links,
+    if not found, raises RequestError"""
+    uri = uri.lstrip('/')
+    if os.path.isdir(os.path.join(ROOT, uri)):
+        body = os.listdir(os.path.join(ROOT, uri))
```

Comment:
You use the value returned by `os.path.join(ROOT, uri)` at least three times in this function.  Would it be worth moving that up into a single local symbol that you can then reference so you don't have to repeat it so much?